packages/p2p/src/components/verification/verification.tsx

```diff
@@ -1,73 +1,102 @@
+    switch (status) {
+        case identity_status_codes.PENDING:
+            return <Localize i18n_default_text='Identity verification in progress.' />;
+        case identity_status_codes.REJECTED:
+            return <Localize i18n_default_text='Identity verification failed. Please try again.' />;
+        case identity_status_codes.VERIFIED:
+            return <Localize i18n_default_text='Identity verification complete.' />;
+        default:
+            return <Localize i18n_default_text='Upload documents to verify your identity.' />;
     }
-
-    if (isMobile() && general_store.should_show_popup) {
-        return <NicknameForm />;
+};
+const getPoaAction = (status: string) => {
+    switch (status) {
+        case document_status_codes.PENDING:
+            return <Localize i18n_default_text='Address verification in progress.' />;
+        case document_status_codes.REJECTED:
+            return <Localize i18n_default_text='Address verification failed. Please try again.' />;
+        case document_status_codes.VERIFIED:
+            return <Localize i18n_default_text='Address verification complete.' />;
+        default:
+            return <Localize i18n_default_text='Upload documents to verify your address.' />;
     }
+};
 
+const Verification = () => {
+    const { general_store } = useStores();
+    const { p2p_poa_required, poa_status, poi_status } = general_store;
+    const redirectToVerification = (route: string) => {
+        const search = window.location.search;
+        let updated_url = `${route}?ext_platform_url=${routes.cashier_p2p}`;
+        if (search) {
+            const url_params = new URLSearchParams(search);
+            const updated_url_params = new URLSearchParams(updated_url);
+            url_params.forEach((value, key) => updated_url_params.append(key, value));
+            updated_url = `${updated_url}&${url_params.toString()}`;
+        }
+        window.location.href = updated_url;
+    };
     const checklist_items = [
         {
-            content: general_store.nickname || <Localize i18n_default_text='Choose your nickname' />,
-            status: general_store.nickname ? 'done' : 'action',
-            onClick: general_store.nickname
-                ? () => {
-                      //do nothing
-                  }
-                : () => {
-                      if (isDesktop()) showModal({ key: 'NicknameModal' });
-                      general_store.toggleNicknamePopup();
-                  },
-        },
-        {
-            content: general_store.poiStatusText(general_store.poi_status),
-            is_disabled: general_store.poi_status !== 'verified' && !general_store.nickname,
-            status: general_store.poi_status === 'verified' ? 'done' : 'action',
+            content: getPoiAction(poi_status),
+            is_disabled: poi_status === identity_status_codes.PENDING,
+            status: poi_status === identity_status_codes.VERIFIED ? 'done' : 'action',
```

Comment:
do we have constants for 'done' and 'action' too? if yes, is it possible to use them?